changelog/12615.bugfix.rst

```diff
@@ -0,0 +1 @@
+Consider entry_points for ``PYTEST_PLUGINS`` and ``pytest_plugins``.
```

Comment:
Could you rewrite this avoiding the imperative mood and including information about the effect it has on the end-users? Also, instead of using `entry_points`(which should've been marked as inline code), use the terminology already present in the docs, and make the link clickable.

Reply:
I can try, but I need some assistance:

- I believe _Consider_ is infinitive, not imperative, but since English is not my primary language I could be easily wrong; how to write _Consider_ without the imperative mood?
- I believe the effect on end users is that both `PYTEST_PLUGINS` and `pytest_plugins` will consider `entry_points.txt` files (maybe I should add `.txt` to the changelog?).  I do not see any other effect on end users.  This is apparently not true, and I miss something important here (that's not surprising since I'm not well familiar with the `pytest` codebase and internals).  Please let me know what other effect for end users do you see here.  Thank you.
- I'm not sure what docs is _the docs_.  Are you talking about [this](https://docs.pytest.org/en/stable/how-to/writing_plugins.html#setuptools-entry-points)?  What terminology should I use instead? _`setuptools`’s entry point_ or something different?
- What link is _the link_?  What exactly do you mean by making a link clickable?

Thank you.

Reply:
>     * I'm not sure what docs is _the docs_.  Are you talking about [this](https://docs.pytest.org/en/stable/how-to/writing_plugins.html#setuptools-entry-points)?  What terminology should I use instead? _`setuptools`’s entry point_ or something different?

I changed it.  Please let me know what do you think about it.

>     * What link is _the link_?  What exactly do you mean by making a link clickable?

I tried to make the `entry points` clickable (copy'n'paste from `writing_plugins.rst`), but I've no idea if I did it properly.  I simply do not know how to verify that.

Reply:
> I believe _Consider_ is infinitive, not imperative, but since English is not my primary language I could be easily wrong; how to write _Consider_ without the imperative mood?

The infinitive has “to” in front of the verb. Without it, it's an instruction, like a TODO-list entry (which is a typically recommended Git commit summary line style). Sometimes, it's useful to use the past tense because the change log is a collection of things that have already changed compared to the previous release (unlike Git commits, which refer to changes since the previous commit and target a completely different audience; hence different writing style).


> I believe the effect on end users is that both `PYTEST_PLUGINS` and `pytest_plugins` will consider `entry_points.txt` files (maybe I should add `.txt` to the changelog?). I do not see any other effect on end users. This is apparently not true, and I miss something important here (that's not surprising since I'm not well familiar with the `pytest` codebase and internals). Please let me know what other effect for end users do you see here. Thank you.

So my problem with this is that it refers to a low-level implementation detail of something external to the project. Entry points are a part of the metadata, and that metadata is usually consumed through a standard interface (`importlib.metadata`) and doesn't really expose to us how that information is stored on disk. We just know that it's connected to some installed distributions. Just a few weeks ago, I was working on getting rid of such low-level details from the docs: https://github.com/pytest-dev/pytest/pull/12469.

The thing is that a change log is targeting end-users, not contributors/devs (except for the corresponding sections). And so it should describe something that users face directly. Do they edit the `entry_points.txt` by hand? Do they pass it to `pytest` somehow? I think not.

We should describe things in the terms that are accessible to an average reader who doesn't really know the subtleties of packaging.

Additionally, I think the word “consider” by itself, might sound a little vague. So I'd replace it with something less generic. Something along the lines of “The plugins specified in the `pytest_plugins` config setting and `PYTEST_PLUGINS` environment variable now allow using entry point names additionally to the importable definitions. Prior to this release, these identifiers used to only work with the `-p` CLI option.”


> * I'm not sure what docs is _the docs_.  Are you talking about [this](https://docs.pytest.org/en/stable/how-to/writing_plugins.html#setuptools-entry-points)?  What terminology should I use instead? _`setuptools`’s entry point_ or something different?

You're referencing the stable docs that have changed a little, as I mentioned above. The diff of the related updates is visible at https://github.com/pytest-dev/pytest/pull/12469 and the latest wording for this specific place is https://docs.pytest.org/en/latest/how-to/writing_plugins.html#making-your-plugin-installable-by-others.


> What link is _the link_? What exactly do you mean by making a link clickable?

In general, all of the mentioned things should be clickable. It could be an entirely external link or a reference using corresponding RST syntax that would link things within the docs on the website and other intersphinx-connected websites.

I think, there are roles for different things that are already documented. Like
```rst
:envvar:`PYTEST_PLUGINS`
```
and
```rst
:globalvar:`pytest_plugins`
```
or
```rst
:ref:`entry points <pip-installable plugins>`
```


> I tried to make the `entry points` clickable (copy'n'paste from `writing_plugins.rst`), but I've no idea if I did it properly. I simply do not know how to verify that.

@mtelka your PR change note preview is visible at https://pytest--12616.org.readthedocs.build/en/12616/changelog.html#bug-fixes

Reply:
> Additionally, I think the word “consider” by itself, might sound a little vague. So I'd replace it with something less generic. Something along the lines of “The plugins specified in the `pytest_plugins` config setting and `PYTEST_PLUGINS` environment variable now allow using entry point names additionally to the importable definitions. Prior to this release, these identifiers used to only work with the `-p` CLI option.”

Done.  Better?

> @mtelka your PR change note preview is visible at https://pytest--12616.org.readthedocs.build/en/12616/changelog.html#bug-fixes

Nice, thanks.

Reply:
Yeah, there's just a few small fixes to make: https://github.com/pytest-dev/pytest/pull/12616/files#r1679811633